src/SWP/Bundle/CoreBundle/Controller/MenuController.php

```diff
@@ -45,16 +43,79 @@ class MenuController extends Controller
      * @Route("/api/{version}/menus/", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_core_list_menu")
      * @Method("GET")
      */
-    public function listAction(Request $request)
+    public function listAction()
     {
         $menuRepository = $this->get('swp.repository.menu');
 
-        $menus = $menuRepository->getPaginatedByCriteria(new Criteria(), [], new PaginationData($request));
+        return new ResourcesListResponse($menuRepository->findRootNodes());
+    }
+
+    /**
+     * Lists all children of menu item.
+     *
+     * @ApiDoc(
+     *     resource=true,
+     *     description="Lists all children of menu item",
+     *     statusCodes={
+     *         200="Returned on success.",
+     *         404="No menus found."
+     *     }
+     * )
+     * @Route("/api/{version}/menus/{id}/children/", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_core_list_children_menu")
+     * @Method("GET")
+     */
+    public function listChildrenAction($id)
+    {
+        $menuRepository = $this->get('swp.repository.menu');
+
+        $menus = $menuRepository->findChildrenAsTree($this->findOr404($id));
 
         return new ResourcesListResponse($menus);
     }
 
     /**
+     * Moves menu item to a specific position.
+     *
+     * @ApiDoc(
+     *     resource=true,
+     *     description="Moves menu item to a specific position",
+     *     statusCodes={
+     *         200="Returned on success.",
+     *         404="No menus found."
+     *     },
+     *     requirements={
+     *         {"name"="id", "dataType"="integer", "requirement"="\d+", "description"="An identifier of Menu item which you want to move"}
+     *     },
+     *     input="SWP\Bundle\MenuBundle\Form\Type\MenuType"
+     * )
+     * @Route("/api/{version}/menus/{id}/move/", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_core_move_menu", requirements={"id"="\d+"})
+     * @Method("PATCH")
+     */
+    public function moveAction(Request $request, $id)
+    {
+        $menuItem = $this->findOr404($id);
+        $form = $this->createForm(MenuItemMoveNodeType::class, [], ['method' => $request->getMethod()]);
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
+            $menuItemManager = $this->get('swp_menu.manager.menu_item');
+            $formData = $form->getData();
+
+            if (isset($formData['parentId'])) {
```

Comment:
for now we have two form fields in order to move menu item:
- parentId
- afterId

If `parentId` is given in request then the menu item will be moved to a position directly under the `parentId` subtree (as a first child of that parentId).
If `afterId` is given, the menu item will be moved after the menu item defined by `afterId`.

It works fine but maybe this should be improved a bit, for example we could use fields like:
- parent - parent menu item under which you want to place moved menu item
- position - position in above parent's subtree under which the given element should be placed. This will require sortable gedmo extension tho, but the form will have full validation of given data class and will stick more to the model itself, because `parentId` and `afterId` properties don't exist in the model. Just an idea for the improvement.